clients/src/test/java/org/apache/kafka/clients/consumer/MockConsumerTest.java

```diff
@@ -100,4 +100,27 @@ public void testConsumerRecordsIsEmptyWhenReturningNoRecords() {
         assertThat(records.isEmpty(), is(true));
     }
 
+    @Test
+    public void testPollDoesNotClearRecordsForPausedPartitions() {
+        TopicPartition partition0 = new TopicPartition("test", 0);
+        TopicPartition partition1 = new TopicPartition("test", 1);
+        consumer.assign(Arrays.asList(partition0, partition1));
+        consumer.addRecord(new ConsumerRecord<>("test", 0, 0, null, null));
+        consumer.addRecord(new ConsumerRecord<>("test", 1, 0, null, null));
+        consumer.updateBeginningOffsets(new HashMap<TopicPartition, Long>() {{
+                put(partition0, 0L);
+                put(partition1, 0L);
+            }});
+        consumer.seekToBeginning(Arrays.asList(partition0, partition1));
+        consumer.assign(Arrays.asList(partition0, partition1));
+        consumer.pause(Collections.singleton(partition1));
+
+        ConsumerRecords<String, String> recordsFirstPoll = consumer.poll(Duration.ofMillis(1));
+        assertThat(recordsFirstPoll.count(), is(1));
```

Comment:
If you want to check that records for non-pause partitions are cleared, we should have a separate test.